### PR TITLE
Adds ability to auto-fix linter errors

### PR DESCRIPTION
**Pull request by anselmbradford**
## Additions

- Adds ability to auto-fix linter errors with a command-line `--fix` flag on the linter task.

## Testing

- Run `gulp lint --fix` and look at your changed files in git. See that `Header.js` and `es5-shim.js` have spacing issues that are automatically fixed.

## Review

- @sebworks 
- @contolini 
- @chosak 
- @jimmynotjim 
- @virginiacc
- @KimberlyMunoz

### CODE REVIEW

**Comment by sebworks** [comment]
This is awesome. 👍

**Comment by chosak** [comment]
Agree, this is very helpful. 👍

**Comment by jimmynotjim** [comment]
Whaaaaa O_O This is amazing!

File: TEST.md [hunk 1]
```diff
@@ -162,3 +162,16 @@ To audit a page's WCAG and Section 508 accessibility:
   4. To test a page aside from the homepage, add the `--u=<path_to_test>` flag.
      For example, `gulp test:a11y --u=contact-us`
      or `gulp test:a11y --u=the-bureau/bureau-structure/`.
+
+# Source code linting
+
+The default test task includes linting of the JavaScript source, build,
+and test files.
+Use the `gulp lint` command from the command-line to run the ESLint linter,
+which defines a number of [rules](http://eslint.org/docs/rules/).
```
Comment (Scotchester) [inline TEST.md]: Suggest changing this line to:

```markdown
which checks the JavaScript against the rules configured in `.eslintrc`.
[See the ESLint docs](http://eslint.org/docs/rules/) for detailed rule descriptions.
```

File: TEST.md [hunk 2]
```diff
@@ -162,3 +162,16 @@ To audit a page's WCAG and Section 508 accessibility:
...
+There are a number of options to the command:
```
Comment (Scotchester) [inline TEST.md]: Start a new paragraph here.

File: gulp/tasks/lint.js [hunk 3]
```diff
@@ -4,16 +4,21 @@ var gulp = require( 'gulp' );
...
 function _genericLint( src ) {
-  return gulp.src( src )
-    .pipe( plugins.eslint() )
+  // Grab the --fix flag from the command-line, if available.
+  var commandLineParams = minimist( process.argv.slice( 2 ) );
+  var willFix = commandLineParams.fix || false;
+  return gulp.src( src, { base: './' } )
+    .pipe( plugins.eslint( { fix: willFix } ) )
     .pipe( plugins.eslint.format() )
+    .pipe( gulp.dest( './' ) )
```
Comment (Scotchester) [inline gulp/tasks/lint.js]: Why is this line added?

File: gulp/tasks/lint.js [hunk 4]
```diff
@@ -4,16 +4,21 @@ var gulp = require( 'gulp' );
...
  * @param {string} src The path to the source JavaScript.
  * @returns {Object} An output stream from gulp.
  */
 function _genericLint( src ) {
-  return gulp.src( src )
-    .pipe( plugins.eslint() )
+  // Grab the --fix flag from the command-line, if available.
+  var commandLineParams = minimist( process.argv.slice( 2 ) );
+  var willFix = commandLineParams.fix || false;
+  return gulp.src( src, { base: './' } )
```
Comment (Scotchester) [inline gulp/tasks/lint.js]: Why is the second parameter added here?

File: TEST.md (hunk 1 above)
Comment (anselmbradford) [inline TEST.md]: Fixed in https://github.com/cfpb/cfgov-refresh/pull/2237/commits/73499da952846a2e8f6e41fe09c96066384bceb2

File: TEST.md (hunk 2 above)
Comment (anselmbradford) [inline TEST.md]: Fixed in https://github.com/cfpb/cfgov-refresh/pull/2237/commits/7a1d73933d8530fe6e68733647ae37dc04a1d539

File: gulp/tasks/lint.js (hunk 3 above)
Comment (anselmbradford) [inline gulp/tasks/lint.js]: So that if the files are auto-fixed they can be written to disk.

File: gulp/tasks/lint.js (hunk 4 above)
Comment (anselmbradford) [inline gulp/tasks/lint.js]: The `base` property is used to set the base path of the source file glob. By setting it to the current directory, and then also providing the same setting in `gulp.dest`, each source file path is used in the destination, effectively overwriting the files. I wasn't sure how else to pass in the destination file paths, which differ widely.

File: gulp/tasks/lint.js (hunk 3 above)
Comment (Scotchester) [inline gulp/tasks/lint.js]: Shouldn't they be written back to `gulp.src`?

Comment (anselmbradford) [inline gulp/tasks/lint.js]: `gulp.src` just reads files as far as I know. `gulp.dest` is what writes them out.

Comment (jimmynotjim) [inline gulp/tasks/lint.js]: @Scotchester `gulp.dest` in this instance isn't the same as the chained variable naming we've been using like `styles.dest`, it's a method for writing to disk.

Comment (Scotchester) [inline gulp/tasks/lint.js]: Gotcha.

**Comment by Scotchester** [comment]
:+1:
